File: CoDrawJavaBackend/src/main/java/com/codraw/CoDraw/dto/CompletedDrawingResponse.java
package com.codraw.CoDraw.dto;

import com.codraw.CoDraw.model.StrokeMessage;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class CompletedDrawingResponse {
    private Long id;
    private String roomCode;
    private String hostUsername;
    private String guestUsername;
    private String roomType;
    private String savedByUsername;
    private Integer strokeCount;
    private LocalDateTime completedAt;
    private List<StrokeMessage> strokes;
}

File: app/src/main/java/com/toan/codraw/data/remote/dto/CompletedDrawingResponse.kt
package com.toan.codraw.data.remote.dto

data class CompletedDrawingResponse(
    val id: Long,
    val roomCode: String,
    val hostUsername: String,
    val guestUsername: String?,
    val roomType: String,
    val savedByUsername: String,
    val strokeCount: Int,
    val completedAt: String,
    val strokes: List<StrokeDto>
)

